muffnn/autoencoder/tests/test_autoencoder.py

```diff
@@ -75,9 +80,38 @@ def _cross_entropy(ytrue, ypred):
     return np.sum(ce, axis=1)
 
 
+class AutoencoderManyEpochs(Autoencoder):
+    """Increase default model capacity and number of epochs so that
+    the default Estimator will pass scikit-learn's sanity checks.
+    """
+    def __init__(self, hidden_units=(256,), batch_size=128, n_epochs=300,
+                 keep_prob=1.0, hidden_activation=tf.nn.relu,
+                 encoding_activation=None,
+                 output_activation=None, random_state=None,
+                 learning_rate=1e-3, loss='mse', sigmoid_indices=None,
+                 softmax_indices=None):
+        super(AutoencoderManyEpochs, self).__init__(
+            hidden_units=hidden_units, batch_size=batch_size,
+            n_epochs=n_epochs, keep_prob=keep_prob,
+            hidden_activation=hidden_activation,
+            encoding_activation=encoding_activation,
+            output_activation=output_activation, random_state=random_state,
+            learning_rate=learning_rate, loss=loss,
+            sigmoid_indices=sigmoid_indices, softmax_indices=softmax_indices,
+        )
+
+
 def test_check_estimator():
     """Check adherence to Estimator API."""
-    check_estimator(Autoencoder)
+    if sys.version_info.major == 3 and sys.version_info.minor == 7:
+        # Starting in Tensorflow 1.14 and Python 3.7, there's one module
+        # with a `0` in the __warningregistry__. Scikit-learn tries to clear
+        # this dictionary in its tests.
+        name = 'tensorboard.compat.tensorflow_stub.pywrap_tensorflow'
+        with mock.patch.object(sys.modules[name], '__warningregistry__', {}):
```

Comment:
Can we not always mock this?  Also, what's the error without this?

Reply:
I would need to add the `mock` backport as a dev dependency for Python 2 if I wanted to always mock it. Since it's not necessary, I wanted to avoid the extra dependency. I could patch only if we have `mock`, but I preferred to make it clear that the problem is very version-limited.

The error from each `test_check_estimator` function is
```
___________________________________________________________________________ test_check_estimator ___________________________________________________________________________

    def test_check_estimator():
        """Check adherence to Estimator API."""
>       check_estimator(MLPRegressorFewerParams)

muffnn/mlp/tests/test_mlp_regressor.py:71:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../anaconda3/envs/muffnn-dev/lib/python3.7/site-packages/sklearn/utils/estimator_checks.py:293: in check_estimator
    check_parameters_default_constructible(name, Estimator)
../../anaconda3/envs/muffnn-dev/lib/python3.7/site-packages/sklearn/utils/estimator_checks.py:2146: in check_parameters_default_constructible
    with ignore_warnings(category=(DeprecationWarning, FutureWarning)):
../../anaconda3/envs/muffnn-dev/lib/python3.7/site-packages/sklearn/utils/testing.py:375: in __enter__
    clean_warning_registry()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    def clean_warning_registry():
        """Clean Python warning registry for easier testing of warning messages.

        We may not need to do this any more when getting rid of Python 2, not
        entirely sure. See https://bugs.python.org/issue4180 and
        https://bugs.python.org/issue21724 for more details.

        """
        reg = "__warningregistry__"
        for mod_name, mod in list(sys.modules.items()):
            if hasattr(mod, reg):
>               getattr(mod, reg).clear()
E               AttributeError: 'int' object has no attribute 'clear'

../../anaconda3/envs/muffnn-dev/lib/python3.7/site-packages/sklearn/utils/testing.py:766: AttributeError
```

The `clean_warning_registry` function is part of scikit-learn's tests.